pkg/storage/s3/s3.go

```diff
@@ -140,6 +141,28 @@ func (d *driver) getS3Service() (*s3.S3, error) {
 
 }
 
+func isBucketNotFound(err interface{}) bool {
+	switch s3Err := err.(type) {
+	case awserr.Error:
+		if s3Err.Code() == "NoSuchBucket" {
+			return true
+		}
+		origErr := s3Err.OrigErr()
+		if origErr != nil {
+			return isBucketNotFound(origErr)
+		}
+	case s3manager.Error:
+		if s3Err.OrigErr != nil {
+			return isBucketNotFound(s3Err.OrigErr)
+		}
+	case s3manager.Errors:
+		if len(s3Err) == 1 {
+			return isBucketNotFound(s3Err[0])
+		}
```

Comment:
Why not loop over the errors, breaking if one of the errors is `NoSuchBucket`?

Reply:
They are different kinds of errors that can be passed in, so only some can be cast to awserr.Error

Reply:
Also, this was copied from the installer.